lib/util/zmqutil.js

```diff
@@ -1,30 +1,211 @@
+
+        , connect: (endpoint) => {
+            try {
+                originalSocket.connect(endpoint)
+                state.endpoints.add(endpoint)
+                log.verbose('Socket connected to:', endpoint)
+            }
+            catch (err) {
+                log.error('Error connecting socket to', endpoint, ':', err)
+
+                if (state.isActive) {
+                    setTimeout(() => {
+                        if (state.isActive) {
+                            log.info('Attempting to reconnect socket to:', endpoint)
+                            wrapper.connect(endpoint)
+                        }
+                    }, state.reconnectDelay)
+                }
+            }
+
+            return wrapper
+        }
+
+        , reconnect: () => {
+            if (!state.isActive) {
+                return wrapper
+            }
+
+            try {
+                originalSocket.close()
+            }
+            catch (err) {
+                log.warn('Error closing socket during reconnect:', err)
+            }
+
+            const newSocket = new SocketClass()
+            Object.assign(originalSocket, newSocket)
+
+            for (const endpoint of state.endpoints) {
+                try {
+                    originalSocket.connect(endpoint)
+                    log.verbose('Socket reconnected to:', endpoint)
+                }
+                catch (err) {
+                    log.error('Error reconnecting socket to', endpoint, ':', err)
+
+                    if (state.isActive) {
+                        setTimeout(() => wrapper.reconnect(), state.reconnectDelay)
+                        break
+                    }
+                }
+            }
+
+            if (type === 'sub' || type === 'pull' || type === 'dealer' || type === 'router' || type === 'reply') {
+                startReceiveLoop()
+            }
+
+            return wrapper
+        }
+
+        , subscribe: (topic) => {
+            if (type === 'sub') {
+                originalSocket.subscribe(
+                    typeof topic === 'string' ? Buffer.from(topic) : topic
+                )
+            }
+
+            return wrapper
+        }
+
+        , unsubscribe: (topic) => {
+            if (type === 'sub') {
+                originalSocket.unsubscribe(
+                    typeof topic === 'string' ? Buffer.from(topic) : topic
+                )
+            }
+            return wrapper
+        }
+
+        , send: (args) => {
+            const sendAsync = async() => {
+                try {
+                    if (Array.isArray(args)) {
+                        await originalSocket.send(args.map(arg =>
+                            Buffer.isBuffer(arg) ? arg : Buffer.from(String(arg))
+                        ))
+                    }
+                    else {
+                        await originalSocket.send(Buffer.isBuffer(args) ? args : Buffer.from(String(args)))
+                    }

```

Comment:
To simplify maybe
```suggestion
                    if (!Array.isArray(args)) {
                        args = [args]
                    }
                    await originalSocket.send(args.map(arg =>
                          Buffer.isBuffer(arg) ? arg : Buffer.from(String(arg))
                    ))

```?